homeassistant/components/vallox/sensor.py

```diff
@@ -223,14 +223,22 @@ class ValloxFilterRemainingSensor(ValloxSensor):
         """Fetch state from the ventilation unit."""
         try:
             days_remaining = int(self._state_proxy.fetch_metric(self._metric_key))
-            days_remaining_delta = timedelta(days=days_remaining)
+            if days_remaining == 0:
+                self._device_class = None
+                self._state = "overdue"
```

Comment:
It's not a good approach to change the device class or state type depending on the state value.

It would be ok to return `None` to mean an unknown state.

Another binary sensor could be added that alerts the user if the filter change is overdue.

Reply:
Thanks for the advice. That was exactly what I also thought about.
Changed it and added it to the branch.